webpack.config.js

```diff
@@ -0,0 +1,47 @@
+const path = require('path')
+const webpack = require('webpack')
+const HtmlWebpackPlugin = require('html-webpack-plugin')
+const CleanWebpackPlugin = require('clean-webpack-plugin')
+
+module.exports = {
+  context: path.resolve(__dirname, 'src'),
+  entry: {
+    app: './app.js'
+  },
```

Comment:
You could write this as `entry: './app.js'`

Reply:
From the [docs](https://webpack.js.org/configuration/entry-context/#naming):

> If a string or array of strings is passed, the chunk is named main. If an object is passed, each key is the name of a chunk, and the value describes the entrypoint for the chunk.

So in order the output to be the file `app.bundle.js` and not `main.bundle.js` an object has to be passed. That's why the `filename` attribute on `output` is `[name].bundle.js`. `[name]` is the object's key from `entry`